src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs

```diff
@@ -19,6 +19,15 @@ public class DirectoryObjectSecurityTests
         private const int WriteAccessMask = 0x02;
         private const int ReadAttributeAccessMask = 0x80;
 
+        [Fact]
+        public void SetCustomDescriptor_Success()
+        {
+            // https://github.com/dotnet/corefx/issues/34151
+            var customObjectSecurity = new CustomDirectoryObjectSecurity();
+            customObjectSecurity.SetSecurityDescriptorSddlForm("D:PAI(A;OICI;0x1200a9;;;BU)");
+            customObjectSecurity.GetAccessRules(true, true, typeof(SecurityIdentifier));
```

Comment:
Can we assert anything?  Validate what GetAccessRules returns?

Reply:
I was trying to focus on the break here. I'll break out what this SDDL is and validate it.